packages/source-iotsitewise/src/asset-modules/util/emitAssetModels.ts

```diff
@@ -0,0 +1,40 @@
+import { DescribeAssetModelResponse } from '@aws-sdk/client-iotsitewise';
+import { SiteWiseAssetDataStreamQuery } from '../../time-series-data/types';
+import { ErrorDetails } from '@iot-app-kit/core';
+import { SiteWiseAssetSession } from '../index';
+
+export const emitAssetModels = ({
+  emit,
+  queries,
+  assetModuleSession,
+  assetModels,
+  errors,
+}: {
+  emit: () => void;
+  queries: SiteWiseAssetDataStreamQuery[];
+  assetModuleSession: SiteWiseAssetSession;
+  assetModels: Record<string, DescribeAssetModelResponse>;
+  errors: Record<string, ErrorDetails>;
+}) => {
+  queries.forEach((query) => {
+    query.assets.forEach((asset) => {
+      assetModuleSession
+        .fetchAssetSummary({ assetId: asset.assetId })
+        .then((assetSummary) => {
+          if (assetSummary && assetSummary.assetModelId != null) {
+            return assetModuleSession.fetchAssetModel({ assetModelId: assetSummary.assetModelId });
+          }
+        })
+        .then((assetModelResponse) => {
+          if (assetModelResponse) {
+            assetModels[asset.assetId] = assetModelResponse;
```

Comment:
Personal pref: fewer mutations would be easier to read (like the composition you're doing for data streams already). Same with `emitAlarms`

Reply:
Agreed, I think we might want to rethink state management in the `subscribeToTimeSeriesData`.